doc/old/best_practices.rst

```diff
@@ -0,0 +1,121 @@
+an error rather than assuming a default behavior or issuing a warning.
+
+This flag does not guarantee strict DICOM results, but is a help for 
+some specific situations that have been added to *pydicom* over time.
+
+In some cases, especially if you are dealing with files that do not 
+strictly adhere to the DICOM Standard, you may need to disable this flag.
+
+To turn on the flag in your code:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.enforce_valid_values = True
+
+Note that you *must not* use 
+:code:`from pydicom.config import enforce_valid_values`.
+That makes the `enforce_valid_values` variable local only to that module,
+so *pydicom* would not see your change to its value.
+
+Future-proofing your code
+-------------------------
+
+*pydicom*, like all software, must balance its evolution with not breaking
+existing code using the library. Sometimes, major changes are necessary
+to make significant improvements to the library.
+
+To help you protect your code against future changes, *pydicom* allows you
+to flag that it should behave as it will for any known upcoming 
+major changes.
+
+Running your code with this turned on will help identify any parts of 
+your code that are not compatible with the known changes in the next major
+version of *pydicom*.
+
+The simplest way to set this behavior is to set an environment variable
+``PYDICOM_FUTURE`` to "True". For example to temporarily turn it on in the
+current terminal session:
+
+.. code-block::
+  
+  SET PYDICOM_FUTURE=True           (Windows)
+
+  export PYDICOM_FUTURE=True        (many linux environments)
+
+If you wish to turn off the behavior, you can
+either remove the environment variable, or set it to "False". See your
+operating system documentation for more details on setting or removing
+environment variables.
+
+The other way to enable the future behavior is to turn it on at run-time
+using the :func:`~pydicom.config.future_behavior` function:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.future_behavior()
+
+If you needed to turn the future behavior off again at run-time, you could
+force re-import of the `config` module:
+
+.. code-block:: python
+
+  from pydicom import config
+  config.enforce_valid_values = True
+  # execute code with flag on...
+  # Reload config in its original state
+  import importlib
+  importlib.reload(config)
+
+.. Note::
+  
+  Re-importing `pydicom.config` will not disable the future behavior if the
+  ``PYDICOM_FUTURE`` environment variable has been set to "True".
+
+
+Limiting the *pydicom* major version in your package
+----------------------------------------------------
+
+Another way to avoid breaking changes in future *pydicom* versions is to
+limit the version of *pydicom* that your code uses.
+
+If you follow standard Python packaging recommendations, you can add a line
+to you `requirements.txt` file to limit the *pydicom* version to the currentmodule
```

Comment:
you > your, currentmodule -> current module?

Reply:
Thanks, fixed.